### PR TITLE
Tab Restore Fixes

### PR DESCRIPTION
**Pull request by kylehickinson**
- [QOL] No longer attempts to track crashes when building the `Fennec` environment.
- [Bug Fix] Updates the tab count properly after restoring

## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [ ] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [x] I have updated the *Unit Tests* to cover new or changed functionality
- [x] I have updated the *UI Tests* to cover new or changed functionality
- [x] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

_None included_

## Notes for testing this patch

_None included_

### CODE REVIEW

File: Client/Application/AppDelegate.swift [hunk 1]
```diff
@@ -117,7 +117,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
 
         browserViewController.restorationIdentifier = NSStringFromClass(BrowserViewController.self)
         browserViewController.restorationClass = AppDelegate.self
-        browserViewController.crashedLastSession = crashedLastSession
+        #if !MOZ_CHANNEL_FENNEC
```
Comment (jhreis) [inline Client/Application/AppDelegate.swift]: Add a comment about _why_ here. (e.g. simulator stop => crash, ads to tons of prompts on sim relaunches)

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -498,10 +498,12 @@ class BrowserViewController: UIViewController {
         let alert = UIAlertController.restoreTabsAlert(
             okayCallback: { _ in
                 self.tabManager.restoreTabs()
+                self.updateTabCountUsingTabManager(self.tabManager, animated: true)
```
Comment (jhreis) [inline Client/Frontend/Browser/BrowserViewController.swift]: Any reason why `restoreTabs()` doesn't trigger a tab count update? This seems very manual / error prone. I know `updateTabCountUsingTabManager` is in different class, but is there some tab creation hook we can leverage?

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -498,10 +498,12 @@ class BrowserViewController: UIViewController {
...
             },
             noCallback: { _ in
                 TabMO.removeAll()
                 self.tabManager.addTabAndSelect()
+                self.updateTabCountUsingTabManager(self.tabManager, animated: true)
```
Comment (jhreis) [inline Client/Frontend/Browser/BrowserViewController.swift]: Basically ditto, regarding `addTabAndSelect`.

**Review by jhreis — CHANGES_REQUESTED** [review]

**Review by iccub — APPROVED** [review]
Looks good to me.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (kylehickinson) [reply Client/Frontend/Browser/BrowserViewController.swift]: It really _should_ be doing it after adding the new tabs, but then fails to do so because it marks itself as "restoring" so the code around doing it fails and the logic around why they have that seems more fragile to remove than to do this.

```swift
if !tabManager.isRestoring {
    updateTabCountUsingTabManager(tabManager)
}
```

What's more confusing is they prefix these lines of code with `// If we are restoring tabs then we update the count once at the end` but actually doesn't update.

Then when it _can_ count it, theres no selected tab yet because the logic is "addTab → selectTab" and selecting a tab doesn't update the tab count because there was no previous tab. That logic is this:

```swift
if let selected = selected, let previous = previous, selected.isPrivate != previous.isPrivate {
    updateTabCountUsingTabManager(tabManager)
}
```

So basically it never updates... But the non-restoring flow calls the `self.updateTabCountUsingTabManager` method.

Comment (jhreis) [reply Client/Frontend/Browser/BrowserViewController.swift]: Something is either broken with the delegate mechanism or some weird timing condition regarding when the delegates are notified:

`tabManager.addDelegate(self)`

```    
func tabManagerDidRestoreTabs(_ tabManager: TabManager) {
     updateTabCountUsingTabManager(tabManager)
}
```

This feels like we're just throwing a bandaid on this, but manually remembering to add tab updates everywhere _will_ break at some point. I'd rather try to fix the setup that is already in place. Either we broke it or inherited some regression.

Comment (kylehickinson) [reply Client/Frontend/Browser/BrowserViewController.swift]: OK I'll give it another pass and figure out what's going on here, removing [needsuplift] for now

**Comment by kylehickinson** [comment]
Updated to not manually update the tab count

**Review by jhreis — APPROVED** [review]
